assets/onboarding/query-string-router/index.jsx

```diff
@@ -0,0 +1,85 @@
+import {
+	useState,
+	useMemo,
+	useContext,
+	createContext,
+} from '@wordpress/element';
+
+import { useEventListener } from '../../react-hooks';
+import { updateRouteURL, getCurrentRouteFromURL } from './url-functions';
+
+/**
+ * Query string router context.
+ */
+const QueryStringRouterContext = createContext();
+
+/**
+ * Query string router component.
+ * It creates a provider with the following values in the context:
+ * - `currentRoute`: The string of the current route.
+ * - `goTo`: Functions that send the user to another route.
+ *
+ * @param {Object} props
+ * @param {string} props.paramName Param used as reference in the query string.
+ * @param {Object} props.children  Render this children if it matches the route.
+ */
+const QueryStringRouter = ( { paramName, children } ) => {
```

Comment:
I'm a bit surprised that there wasn't a standard component we could use for this. Did we look around for what others used? Does Woo use their own custom router?

Reply:
Actually the React Router wouldn't work well for us because we are routing based in the queryString because we wouldn't have how to make a URL rewrite in the server for the users.

Another approach would be to work with hash (window.location.hash). But it's not so usual and WooCommerce also works with query string.

And checking how WooCommerce did it:

> I checked how WooCommerce routes the pages in their wizard and it uses the react-router with `@woocommerce/navigation`. For now, I decided don't add these dependencies and create a simple router to work based in query string. I decided it because the code that we would need to write for both cases would be almost the same. And writing a simple implementation, we can evolve how we need and we don't have more dependencies in the project.

Reply:
Sounds good, thanks